src/query/ts/metadata.go

```diff
@@ -20,22 +20,51 @@
 
 package ts
 
-// MetricType is the enum for metric types.
-type MetricType int
+// M3MetricType is the enum for M3 metric types.
+type M3MetricType uint8
 
 const (
-	// MetricTypeGauge is the gauge metric type.
-	MetricTypeGauge MetricType = iota
+	// M3MetricTypeGauge is the gauge metric type.
+	M3MetricTypeGauge M3MetricType = iota
 
-	// MetricTypeCounter is the counter metric type.
-	MetricTypeCounter
+	// M3MetricTypeCounter is the counter metric type.
+	M3MetricTypeCounter
 
-	// MetricTypeTimer is the timer metric type.
-	MetricTypeTimer
+	// M3MetricTypeTimer is the timer metric type.
+	M3MetricTypeTimer
+)
+
+// PromMetricType is the enum for Prometheus metric types.
+type PromMetricType uint8
+
+const (
+	// PromMetricTypeUnknown is the unknown Prometheus metric type.
+	PromMetricTypeUnknown PromMetricType = iota
```

Comment:
nit; can remove `MetricType` from the name, so these would be `PromCounter` and `M3Counter`?

Reply:
Started doing it, and then realized that names such as `PromUnknown`and `PromInfo` would lose too much context. Decided to keep the current names, also consistent with the preexisting `M3MetricTypeGauge` etc.